src/Plutip/Server.purs

```diff
@@ -95,26 +100,40 @@ withPlutipContractEnv
   -> Aff a
 withPlutipContractEnv plutipCfg distr cont =
   withPlutipServer $
-    withPlutipCluster \response ->
-      withWallets response \wallets ->
-        withPostgres response
-          $ withOgmios response
-          $ withOgmiosDatumCache response
-          $ withCtlServer
-          $ withContractEnv (flip cont wallets)
+    withPlutipCluster \(ourKey /\ response) ->
+      withPostgres response
+        $ withOgmios response
+        $ withOgmiosDatumCache response
+        $ withCtlServer
+        $ withContractEnv \env ->
+            withWallets env ourKey response (cont env)
   where
   withPlutipServer :: Aff a -> Aff a
   withPlutipServer =
     withResource (startPlutipServer plutipCfg) stopChildProcess <<< const
 
-  withPlutipCluster :: (ClusterStartupParameters -> Aff a) -> Aff a
-  withPlutipCluster cc = withResource (startPlutipCluster plutipCfg distr)
+  withPlutipCluster
+    :: ((PrivatePaymentKey /\ ClusterStartupParameters) -> Aff a) -> Aff a
+  withPlutipCluster cc = withResource
+    ( startPlutipCluster plutipCfg
+        ( [ BigInt.fromInt 1_000_000_000, BigInt.fromInt 1_000_000_000 ] /\
```

Comment:
We should insert the extra keys elsewhere. And we should calculate our amounts based on requested amounts.